### ISSUE TITLE
[apex] Apex files ending in "Test" are skipped with a number of rules

### ISSUE THREAD

**Issue by stephen-carter-at-sf** [pmd/pmd#5386]
**Affects PMD Version:**  From 6.x+ to 7.x+

Make sure, to test with the latest PMD version.

**Description:**

Users of the Salesforce Code Analyzer are complaining that their source files (which happen to end in "Test" but are not unit test files) are getting skipped for rules like ApexCRUDViolation and ApexSharingViolations.

Specifically, looking at the source code it looks like https://github.com/pmd/pmd/blob/main/pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/internal/Helper.java#L54 has
 return className.endsWith("Test");
 
 But this shouldn't be an indication of Test files should it? I think a test must have an `@isTest` annotation doesn't it?
 
 Is this to handle really old versions of apex or something? Can we please remove this so that the rules run on classes that end in "Test" when they don't have an `@isTest` annotation?

**Exception Stacktrace:**

```
# Copy-paste the stack trace here
```

**Code Sample demonstrating the issue:**

Rule doesn't run on:
```java
public class CORE_PGTest{

public void test(){
List<Account> lsAcct = [select ID FROM Account ORDER BY Name ASC limit 2];

List<Account> lsAcct2 = new List<Account>();
lsAcct2 = [select id, Type from account limit 3];

}
}
```

but rule does run on:
```java
public class CORE_PGtest{

public void test(){
List<Account> lsAcct = [select ID FROM Account ORDER BY Name ASC limit 2];

List<Account> lsAcct2 = new List<Account>();
lsAcct2 = [select id, Type from account limit 3];

}
}
```

**Comment by adangel** [comment]
Thanks for reporting. I'm not a salesforce developer, so I can't answer your questions about "But this shouldn't be an indication of Test files should it? I think a test must have an @isTest annotation doesn't it? Is this to handle really old versions of apex or something?". 

The suffix check was added with commit https://github.com/pmd/pmd/commit/a08b478f099606344c36240e841514adb49938a4 a couple of years ago. The commit message doesn't tell much, just "Legacy test class declaration".

So, the question now, to all who know salesforce apex: Is a legacy test class declaration still a thing today?

> Can we please remove this so that the rules run on classes that end in "Test" when they don't have an @isTest annotation?

Yes, we can remove this, if this legacy test class declaration is not valid anymore/has no usages in the apex world.

**Comment by stephen-carter-at-sf** [comment]
Let me ask the Salesforce Apex team about this one and have them report back here.

**Comment by stephen-carter-at-sf** [comment]
OK The Salesforce Apex team got back to me and said:
> if it is not marked @isTest then it's not a test and should be analyzed

So can the PMD folks remove that check and simply rely on the other test related markers?

**Comment by dschach** [comment]
@stephen-carter-at-sf I opened a PR to force PMD to rely only on the `@IsTest` annotation and to ignore the class name.
